### PR TITLE
Update debug per 5.0.0 changes.

### PR DESCRIPTION
**Pull request by JamesDrummond**
Signed-off-by: James Drummond jdrummond@codenvy.com

### CODE REVIEW

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 1]
```diff
@@ -187,3 +191,55 @@ Among other features, it also has the capability to trigger a debug session:
...
+
+1. Go to `Run > Edit Debug Configurations` from the main menu
+1. Create a new `NODEJS` configuration
+1. Change any settings if necessary. The defaults are usually OK
+1. Click the `Debug` button
+1. The debugger will break at first line of code
+
+![debug-nodejs-config.png]({{ base }}{{ site.links["debug-nodejs-config.png"] }})
+
+### Creating a Command to View the Preview URL
```
Comment (bmicklea) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Is this section necessary?

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 2]
```diff
@@ -187,3 +191,55 @@ Among other features, it also has the capability to trigger a debug session:
...
+
+{{site.product_formal_name}}'s workspaces have machine(s) that are docker container(s). These workspace machines have exposed ports which are initially set by [stack]({{base}}{{site.links["ws-stacks"]}}) and [additional exposed ports](#exposed-ports) can be added through the [machine information in dashboard]({{base}}{{site.links["ws-machines"]}}#dashboard-machine-information)
+
+1. Add a command `Run > Edit Commands` 
+1. Give the command a name like "View Preview URL"
+1. Add a fictitious command `echo` for required command line
+1. Provide the preview URL for your app such as `http://${server.port.<port>}/`
+
+### Using the Node.js Debugger
```
Comment (bmicklea) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: This seems like it duplicates L201-L204 - is it needed?

File: src/main/_docs/use-che-as-an-ide/ide-debug.md [hunk 3]
```diff
@@ -187,3 +191,55 @@ Among other features, it also has the capability to trigger a debug session:
...
+1. Create a new `NODEJS` configuration
+1. Change any settings if necessary. The defaults are usually OK
+1. Click the `Debug` button
+1. The debugger will break at first line of code
+
+![debug-nodejs-config.png]({{ base }}{{ site.links["debug-nodejs-config.png"] }})
+
+### Creating a Command to View the Preview URL
+
+{{site.product_formal_name}}'s workspaces have machine(s) that are docker container(s). These workspace machines have exposed ports which are initially set by [stack]({{base}}{{site.links["ws-stacks"]}}) and [additional exposed ports](#exposed-ports) can be added through the [machine information in dashboard]({{base}}{{site.links["ws-machines"]}}#dashboard-machine-information)
```
Comment (bmicklea) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: This mostly duplicates the ## Exposed Ports section at the bottom maybe we should just refer to that section instead?

**Comment by bmicklea** [comment]
@JamesDrummond I made some tweaks to the file but it feels like it's a bit repetitive. If you want I can trim out the extraneous sections, let me know.

File: src/main/_docs/use-che-as-an-ide/ide-debug.md (hunk 3 above)
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Updated.

File: src/main/_docs/use-che-as-an-ide/ide-debug.md (hunk 1 above)
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: Think so. Should make more sense with revisions.

File: src/main/_docs/use-che-as-an-ide/ide-debug.md (hunk 2 above)
Comment (JamesDrummond) [inline src/main/_docs/use-che-as-an-ide/ide-debug.md]: This section is to add a more in-depth instructions.

**Comment by JamesDrummond** [comment]
Moved exposed ports to tutorial doc.

**Pull request merged**